scaleway/provider.go

```diff
@@ -365,5 +378,10 @@ func loadProfile(ctx context.Context, d *schema.ResourceData) (*scw.Profile, err
 			tflog.Debug(ctx, fmt.Sprintf("cannot guess region: %s", err.Error()))
 		}
 	}
+
+	// If any projectID is found on environment or on profiles we return error
+	if !terraformTestCassettesEnabled && (profile.DefaultProjectID == nil || len(*profile.DefaultProjectID) == 0) {
+		return nil, fmt.Errorf("the project_id is not found in the environment or the profile configurations")
+	}
```

Comment:
This should be enough to check after profile merge, previous warnings seems unnecessary.
Also, some resources does not require a project_id